### DISCUSSION TITLE
Best practice about saving FCM tokens to backend with auth

### DISCUSSION THREAD

**Discussion by slavko-lukic** [invertase/react-native-firebase#7465] [Q&A]
I ran into small issue when I'm trying to save FCM token to backend but when I have authentication implemented. 
I will demonstrate on the example from docs but alter it a bit to show you what I mean.

```javascript
import React, { useEffect } from 'react';
import messaging from '@react-native-firebase/messaging';

async function saveTokenToDatabase(token) {
  // here we save FCM to database on our server which requires JWT
}

function App() {
  const { isAuthenticated } = useAuth()
  
  useEffect(() => {
    if (!isAuthenticated) return;
  
    // Get the device token
    messaging()
      .getToken()
      .then(token => {
        return saveTokenToDatabase(token);
      });

    // Listen to whether the token changes
    return messaging().onTokenRefresh(token => {
      saveTokenToDatabase(token);
    });
  }, [isAuthenticated]);
}
```

In example above we don't want to try and save FCM token to backend because request will fail if JWT is not present, so because of that I put guard statement at the beginning of the useEffect to prevent FCM token handling if user is not authenticated.

Problem arises when user logs out which clears the FCM token and logs back in.
Then `saveTokenToDatabase` gets called twice, first time here:
```javascript
messaging()
      .getToken()
      .then(token => {
        return saveTokenToDatabase(token);
      });
```
and second time in `onTokenRefresh`.

What is the best way to handle this?

**Comment by slavko-lukic** [comment]
Seems to me that I can remove `onTokenRefresh` completely because of the way I implemented it, my backend should always have newest token.
What else could trigger `onTokenRefresh` while my app is open except calling `messaging().deleteToken()`?
If I delete tokens at logout but request them again at login with `getToken()` one case should be covered.

**Reply by mikehardy** [reply]
I think over a long enough period of time the auth token may expire and be auto-refreshed? the FCM token would be the same though I believe
